swap_meet/vendor.py

```diff
@@ -0,0 +1,112 @@
+
+#wave_1
+class Vendor:
+    def __init__(self, inventory = None):
+        if inventory is None:
+            self.inventory = []
+        else: 
+            self.inventory = inventory 
+        
+    def adding_to_inventory(Vendor): #checks item in inventory & returns True or False 
+        if item in inventory:
+            self.inventory.add(item)
+        else:
+            self.inventory.remove(item)
+        return False 
+    
+    def add(self, item): #append item in inventory 
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item): #remove item from inventory
+        if item in self.inventory:
+            self.inventory.remove(item)
+        else:
+            return False 
+            
+        return item 
+
+#wave_2 
+    def get_by_category(self, category):
+        """
+        gets items by category
+        input: inventory from items
+        output: list of category 
+
+        """
+        item_by_category = []
+        for item in self.inventory:
+            if item.category == category:
+                item_by_category.append(item)
+        return item_by_category
+
+#wave_3
+    def swap_items(self, friend_vendor, my_item, their_item):
+        """
+        check inventory items of each Vendor
+        input: adds and removes items from each Vendor inventory 
+        output: Return True or False 
+        """
+        if my_item in self.inventory and their_item in friend_vendor.inventory:
```

Comment:
Consider refactoring to check the edge case first (similar to the comment for `remove`